### DISCUSSION TITLE
Share state between events

### DISCUSSION THREAD

**Discussion by Dregorio1** [hyprland-community/hyprland-rs#247] [Q&A] [answered]
Hello, how can I share state between events. Like I want to create in `main` function vector where I can store data from one async event and use it in another? Any code snippet will be welcome.

**Comment by yavko** [comment, marked as answer]
> can I share state between events. Like I want to create in `main` function vector where I can store data from one async event and use it in another? Any code snippet will be welcome.

With the limited information you've provided, I think the best way to do this would be to use a static Mutex if using a AsyncEventListener, maybe like so. I haven't tested it, but it should compile.

```rs
use hyprland::event_listener::EventListener;
use std::sync::Mutex;

static TEST_THING: Mutex<Vec<String>> = Mutex::new(Vec::new());

#[tokio::main]
async fn main() -> hyprland::Result<()> {
    let mut listener = EventListener::new();
    listener.add_workspace_change_handler(move |workspace| {
        TEST_THING.lock().unwrap().push(workspace.to_string());
        println!("vec: {:?}", TEST_THING);
    });
    listener.start_listener_async().await?;
    Ok(())
}

```
